src/main/java/org/akhq/modules/AuditModule.java

```diff
@@ -13,12 +17,11 @@
 import org.apache.kafka.clients.producer.RecordMetadata;
 
 import java.time.LocalDateTime;
+import java.util.Optional;
 
 @Singleton
 @Slf4j
 public class AuditModule {
-
-    @Inject
     SecurityService securityService;
```

Comment:
Alternatively you can make this bean Optional and then just check the field before setting the username, it makes this a very small change to the code and doesn't require messing with application context:
```java
public class AuditModule {
    @Inject
    Optional<SecurityService> securityService;
    ...

    public void save(AuditEvent event) {
        ...
        byte[] value;
        securityService.ifPresent(ss -> ss.username().ifPresent(event::setUserName));
        ...
    }
}
```

Reply:
That's definitely better, I'll refactor with your suggestion.